File: Low Level Design (LLD)/Problems/07. SocialApp/Cpp/src/Story.cpp
#include "Story.h"
#include <chrono>

Story::Story(int id, std::shared_ptr<User> author, const std::string& content)
    : id_(id), author_(author), content_(content),
      createdAt_(std::chrono::system_clock::now()) {}

int Story::getId() const { return id_; }
std::shared_ptr<User> Story::getAuthor() const { return author_; }
const std::string& Story::getContent() const { return content_; }

bool Story::isExpired() const {
    auto now = std::chrono::system_clock::now();
    return now - createdAt_ > std::chrono::hours(24);
}

File: Low Level Design (LLD)/Problems/07. SocialApp/Java/models/Story.java
package models;

import java.time.Duration;
import java.time.Instant;

public class Story {
    private final int id;
    private final User author;
    private final String content;
    private final Instant createdAt;

    public Story(int id, User author, String content) {
        this.id = id;
        this.author = author;
        this.content = content;
        this.createdAt = Instant.now();
    }

    public boolean isExpired() {
        return Duration.between(createdAt, Instant.now()).toHours() > 24;
    }

    // Getters
    public int getId() {
        return id;
    }

    public User getAuthor() {
        return author;
    }

    public String getContent() {
        return content;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }
}
